src/index.ts

```diff
@@ -102,14 +107,70 @@ export default class IndentTune implements BlockTune {
         if (this.config.multiblock && !this.config.tuneName)
             console.error("IndentTune config 'tuneName' was not provided, this is required for multiblock option to work.")
 
-        window.addEventListener("resize", (e) => this.onResize.call(this, e));
+        window.addEventListener('resize', (e) => this.onResize.call(this, e))
+
+        if (this.config.autoIndent) {
+            this.handleAutoIndent()
+        }
     }
 
     // prepare?(): void | Promise<void> {
     // }
     // reset?(): void | Promise<void> {
     // }
 
+    private handleAutoIndent() {
+        // Initialise MutationObserver
+        const observer = new MutationObserver((mutationsList) => {
+            for (const mutation of mutationsList) {
+                if (mutation.type === 'childList') {
+                    this.autoIndentAddedBlocks(mutation, observer)
+                }
+            }
+        })
+
+        // Start observing
+        observer.observe(document.body, {
+            childList: true,
+            subtree: true,
+        })
+    }
+
+    private autoIndentAddedBlocks(mutation: MutationRecord, observer: MutationObserver) {
+        for (let i = 0; i < mutation.addedNodes.length; i++) {
+            const node = mutation.addedNodes[i]
+            if (node instanceof Element && node.classList.contains('ce-block') && node.getAttribute('data-id') === this.block?.id) {
+                observer.disconnect()
```

Comment:
Couldn't all this observer logic be simplified to adding 'this.indentCurrentBlockAutomaticallyIfNeeded()' in the constructor and compare previous sibling to know if yes or no?